### PR TITLE
Bug Fix - metrics.py

### PR DESCRIPTION
**Pull request by ashok-arjun**
Turning on evaluation mode with `model.eval()` in `get_model_size` may cause a lot of errors if the user does not switch it back to the original training state in the calling function.

I've integrated restoring the previous state here.

### CODE REVIEW

**Comment by arthurdouillard** [comment]
Indeed, you're right.

Thanks for the bug fix!
